handler/src/main/java/io/netty/handler/ssl/JdkAlpnSslEngine.java

```diff
@@ -62,12 +60,15 @@ private static void updateAvailability() {
                     "protocolSelector");
             ALPN.put(engine, new ServerProvider() {
                 @Override
-                public String select(List<String> protocols) {
+                public String select(List<String> protocols) throws SSLException {
                     try {
                         return protocolSelector.select(protocols);
+                    } catch(SSLException e) {
+                        throw e;
                     } catch (Throwable t) {
-                        PlatformDependent.throwException(t);
-                        return null;
+                        // Ensure that all exceptions are propagated as SSLExceptions
+                        // so that the SslHandler properly fails the handshake.
+                        throw new SSLException(t);
```

Comment:
It looks like the JDK will re-wrap it as a `SSLException`: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/sun/security/ssl/Handshaker.java/#1344

If we throw a `SSLHandshakeException` this wouldn't happen but I don't know if it matters much.